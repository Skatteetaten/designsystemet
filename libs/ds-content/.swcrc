{
  "jsc": {
    "target": "es2021",
    "parser": {
      "syntax": "typescript",
      "tsx": true,
      "decorators": true,
      "dynamicImport": true
    },
    "transform": {
      "react": {
        "runtime": "automatic",
        "importSource": "react"
      },
      "decoratorMetadata": true,
      "legacyDecorator": true
    },
    "keepClassNames": true,
    "externalHelpers": true,
    "loose": true
  },
  "module": {
    "type": "es6"
  },
  "sourceMaps": true,
  "exclude": ["jest.config.ts",".*\\.spec.tsx?$",".*\\.test.tsx?$","./src/jest-setup.ts$","./**/jest-setup.ts$",".*.js$"]
}
