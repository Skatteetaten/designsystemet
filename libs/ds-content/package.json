{
  "name": "@skatteetaten/ds-content",
  "groupId": "no.skatteetaten.aurora",
  "version": "1.4.0",
  "license": "Apache-2.0",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "type": "module",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "react": "^18",
    "@skatteetaten/ds-core-utils": "1.4.0",
    "@skatteetaten/ds-typography": "1.4.0",
    "@skatteetaten/ds-icons": "1.4.0",
    "react-i18next": "^14",
    "i18next": "^23"
  },
  "scripts": {},
  "dependencies": {},
  "devDependencies": {}
}
