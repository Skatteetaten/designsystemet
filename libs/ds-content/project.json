{
  "name": "ds-content",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-content/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-content",
        "tsConfig": "libs/ds-content/tsconfig.lib.json",
        "project": "libs/ds-content/package.json",
        "entryFile": "libs/ds-content/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-content/rollup.config.js"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-content/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/ds-content/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-content/**/*.css",
          "libs/ds-content/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-content"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-content {args.switches}"
      }
    }
  }
}
