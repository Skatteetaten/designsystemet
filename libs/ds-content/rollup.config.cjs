const path = require('path');

const createRollupConfig = require('../ds-dev-config/build/rollup-config-factory.cjs');

const srcDir = path.resolve(__dirname, `src`);
const pathToTsConfig = 'libs/ds-content/tsconfig.lib.json';
const pathToCSS = path.resolve(__dirname, `**/*.{css,scss}`);
const outputDir = path.resolve('./', `dist/libs/ds-content/`);

module.exports = (config) =>
  createRollupConfig(config, srcDir, pathToTsConfig, pathToCSS, outputDir);
