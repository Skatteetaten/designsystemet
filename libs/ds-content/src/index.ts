export * from './Card/Card';
export * from './Card/Card.types';
export * from './CardAlert/CardAlert.types';
export * from './DescriptionList/DescriptionList';
export * from './DescriptionList/DescriptionList.types';
export * from './Panel/Panel';
export * from './Panel/Panel.types';
