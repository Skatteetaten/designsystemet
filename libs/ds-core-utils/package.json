{
  "name": "@skatteetaten/ds-core-utils",
  "groupId": "no.skatteetaten.aurora",
  "version": "1.1.1",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^14"
  },
  "scripts": {}
}
