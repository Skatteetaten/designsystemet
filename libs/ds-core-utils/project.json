{
  "name": "ds-core-utils",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-core-utils/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-core-utils",
        "tsConfig": "libs/ds-core-utils/tsconfig.lib.json",
        "project": "libs/ds-core-utils/package.json",
        "entryFile": "libs/ds-core-utils/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-core-utils/rollup.config.cjs"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-core-utils/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/ds-core-utils"],
      "options": {
        "jestConfig": "libs/ds-core-utils/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-core-utils/**/*.css",
          "libs/ds-core-utils/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-core-utils"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-core-utils {args.switches}"
      }
    }
  }
}
