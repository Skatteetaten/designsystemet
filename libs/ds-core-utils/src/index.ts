export * from './SkatteetatenLogo/SkatteetatenLogo';
export * from './base-props.types';
export * from './defaults';
export * from './ExternalLayout/ExternalLayout';
export * from './InternalLayout/InternalLayout';
export * from './i18n';
export * from './Separator/Separator';
export * from './hooks/hooks';
export * from './formatter/formatter';
