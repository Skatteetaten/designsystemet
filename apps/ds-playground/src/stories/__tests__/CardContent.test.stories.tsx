import { Meta, StoryObj } from '@storybook/react';
import { expect, within } from '@storybook/test';

import { Card } from '@skatteetaten/ds-content';
import { statusArr } from '@skatteetaten/ds-core-utils';

const meta = {
  component: Card.Content,
  title: 'Tester/Card/Content',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: {
      table: { disable: true },
      control: 'text',
    },
    rightContent: {
      table: { disable: true },
      control: 'text',
    },
    spacing: {
      table: { disable: true },
    },
  },
} satisfies Meta<typeof Card.Content>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultText =
  'Avvist av kortutsteder. Ta kontakt med kortutsteder for mer informasjon.';

const defaultArgs = {
  children: defaultText,
  variant: statusArr[1],
  title: 'Merknad',
};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLDivElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    await expect(container).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlId',
    className: 'dummyClassname',
    lang: 'en',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const cardNote = canvas.getAllByRole('generic')[1];
    await expect(cardNote).toHaveAttribute('id', 'htmlId');
    await expect(cardNote).toHaveClass('dummyClassname');
    await expect(cardNote).toHaveAttribute('lang', 'en');
    await expect(cardNote).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;
