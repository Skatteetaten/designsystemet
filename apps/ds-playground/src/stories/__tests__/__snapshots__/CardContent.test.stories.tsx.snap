// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Card/Content WithAttributes play-test 1`] = `
<div data-test-block="true">
  <div id="htmlId"
       class="_cardContent_48t19_1 dummyClassname"
       lang="en"
       data-testid="123ID"
  >
    <div>
      Gruppering av opplysninger i skjema, for eksempel inntekter, personer eller oppsummeringer.
    </div>
  </div>
</div>
`;

exports[`Tester/Card/Content WithChildrenRigthContent smoke-test 1`] = `
<div data-test-block="true">
  <div class="_card_nrlm6_5"
       data-color="white"
  >
    <div class="_cardContent_48t19_1">
      <div>
        <ul class="_list_1xbs1_1 _list_withBullet_1xbs1_8">
          <li class="_element_1b7zf_1">
            første element
          </li>
          <li class="_element_1b7zf_1">
            andre element
          </li>
        </ul>
      </div>
      <div class="_rightContent_48t19_51">
        <div class="_tag_1ciua_1 _tag_ochre_1ciua_22 _tag_medium_1ciua_34">
          <div class="_iconWrapper_1ciua_12"
               aria-hidden="true"
          >
            <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_1ciua_12"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M11 9h2V7h-2v2Zm1 11c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm-1 15h2v-6h-2v6Z">
              </path>
            </svg>
          </div>
          Endret av deg
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/Card/Content WithCustomClassNames play-test 1`] = `
<div data-test-block="true">
  <div class="_cardContent_48t19_1">
    <div class="dummyClassname">
      Gruppering av opplysninger i skjema, for eksempel inntekter, personer eller oppsummeringer.
    </div>
    <div class="_rightContent_48t19_51 dummyClassname">
      høyrejustert innhold
    </div>
  </div>
</div>
`;

exports[`Tester/Card/Content WithRef play-test 1`] = `
<div data-test-block="true">
  <div class="_cardContent_48t19_1"
       id="dummyIdForwardedFromRef"
  >
    <div>
      Gruppering av opplysninger i skjema, for eksempel inntekter, personer eller oppsummeringer.
    </div>
  </div>
</div>
`;
